x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/services/package_installer/package_installer.ts

```diff
@@ -146,7 +151,11 @@ export class PackageInstaller {
         productVersion,
       });
 
-      await ensureDefaultElserDeployed({ client: this.esClient });
+      if (this.elserInferenceId === defaultInferenceEndpoints.ELSER) {
+        await ensureDefaultElserDeployed({
+          client: this.esClient,
+        });
+      }
```

Comment:
Should we still make sure the model is deployed?

Reply:
I mean you have a condition to only check the default version of ELSER. Did you consider changing `ensureDefaultElserDeployed` method to `ensureElserDeployed` and pass the inference_id?

Reply:
I mean that even if we pass custom inference_id it's not going to work, because if `pt_tiny_elser` was deleted then we won't be able to download it as we are in case of default ELSER